docs/src/reference/webhooks.md

```diff
@@ -0,0 +1,188 @@
+  "node_id": <node_id>,
+  "event_type": <event_type>
+}
+```
+
+## Adding a webhook to a greenlight node
+
+### Prerequisites
+- A public tls-secured endpoint 
+- Access to a greenlight node's device certificate
+
+To add a webhook for a greenlight node we first need to initialize a scheduler using the node id and device certificate (returned in the node registration response).
+
+=== "Rust"
+	```rust
+	let device_cert = include_bytes!("path-to-device-cert");
+	let device_key = include_bytes!("path-to-device-key");
+
+	let credentials = Builder::as_device()
+		.with_identity(device_cert, device_key)
+		.build()
+		.expect("Failed to build Device credentials");
+
+	let node_id = hex::decode("hex-node-id").unwrap();
+
+	let scheduler = Scheduler::with_credentials(
+		node_id,
+		gl_client::bitcoin::Network::Bitcoin,
+		utils::scheduler_uri(),
+		credentials
+	)
+	.await
+	.unwrap();
+	```
+
+=== "Python"
+	```python
+	from pathlib import Path
+	from glclient import Credentials, TlsConfig
+
+	certpath = Path("device.pem")
+	keypath = Path("device-key.pem")
+	capath = Path("ca.pem")
+	runepath = Path("rune")
+
+	creds = Credentials.from_parts(
+		certpath.open(mode="rb").read(),
+		keypath.open(mode="rb").read(),
+		capath.open(mode="rb").read(),
+		runepath.open(mode="rb").read(),
+	)
+	
+	node_id = bytes.fromhex("hex-node-id")
+	
+	scheduler = scheduler = Scheduler(
+		node_id=node_id,
+		network="bitcoin",
+		creds=creds
+	)
+	```
+
+Once we're able to initialize the scheduler, we simply need to call `add_outgoing_webhook` with a well-formed url to finish adding the webhook. 
+
+!!! warning "Don't forget to secure your webhook secrets"
+
+	Notice that we call `save_secret_to_db` to save the secret needed to validate webhook requests for this node. This secret cannot be recovered if lost and must be securely stored immediately after the addition of the webhook.
+
+=== "Rust"
+	```rust
+	use gl_client::scheduler::Scheduler;
+	use gl_client::bitcoin::Network;
+
+	let webhook_uri = "https://example.com";
+	let add_webhook_response = scheduler.add_outgoing_webhook(webhook_uri).await.unwrap();
+
+	save_secret_to_db(signer.node_id(), &add_webhook_response.secret);
+	```
+
+=== "Python"
+	```python
+	from glclient import Scheduler
+	
+	scheduler = Scheduler(
+		node_id=signer.node_id(),
+		network="bitcoin",
+		tls=tls,
+	)
+	
+	webhook_uri = "https://example.com"
+	scheduler.add_outgoing_webhook(webhook_uri)
+
+	save_secret_to_db(signer.node_id(), add_webhook_response.secret);
+	```
+
+## Verifying webhook payloads
+
+Webhook payloads can be verified using the secret returned from `add_outgoing_webhook`. The secret can not be shared across nodes and is only valid for the node it was returned for. The secret serves as the key needed to validate the payload in an hmac-sha256 hash. If the payload is valid, the resulting hash should be equal to the base58-encoded value contained within the 'gl-signature' header of the request sent to the webhook.
```

Comment:
We might need to specify that the payload must be hashed and signature verified without parsing the JSON and then serializing again, as the whitespace in the JSON is not guaranteed to be reproduced faithfully. A middleware would be perfect here, but we just need to ensure the webhook server and the recipient of the webhook call agree on the cleartext down to the byte.

Reply:
Done